ql/src/java/org/apache/hadoop/hive/ql/optimizer/SharedWorkOptimizer.java

```diff
@@ -1791,9 +1792,16 @@ private static boolean validPreConditions(ParseContext pctx, SharedWorkOptimizer
     // If we do, we cannot merge. The reason is that Tez currently does
     // not support parallel edges, i.e., multiple edges from same work x
     // into same work y.
+    EdgePredicate edgePredicate;
+    if (pctx.getConf().getBoolVar(ConfVars.HIVE_SHARED_WORK_PARALLEL_EDGE_SUPPORT)) {
+      edgePredicate = new EdgePredicate(EnumSet.<EdgeType> of(EdgeType.DPP, EdgeType.SEMIJOIN, EdgeType.BROADCAST));
+    } else {
+      edgePredicate = new EdgePredicate(EnumSet.<EdgeType> of(EdgeType.DPP));
```

Comment:
If I understand correctly the old behavior didn't allow `DPP` edge types. Should we follow that if  `hive.optimize.shared.work.parallel.edge.support` is false?

Reply:
yes; the old behaviour was to not allow even those - however because DPP edges are not physical tez edges they can be anywhere without restrictions; but I could remove DPP as well to be extra safe